pages/vulnerabilities/Unrestricted_File_Upload.md

```diff
@@ -369,7 +369,7 @@ And some special recommendations for the developers and webmasters:
     name of a file plus its extension should be less than 255 characters
     (without any directory) in an NTFS partition.
   - It is recommended to use an algorithm to determine the filenames.
-    For instance, a filename can be a MD5 hash of the name of file plus
+    For instance, a filename can be a Keccak hash of the name of file plus
```

Comment:
Why not SHA3?

Even better why not remove the algo altogether and just leave "a hash" then no future update is necessary.

Reply:
Actually, that's what I meant (isn't it known as SHA-3?) Yeah, it's better to leave just "a hash"

Reply:
😀 you using "keccak" actually made me go read about this. Apparently when NIST adopted Keccak they actually changed the padding slightly so while it's the same algo it uses a different padding and so it's technically not the exact same :shrug: Learn something everyday 😄

Reply:
```suggestion
    For instance, a filename can be a hash of the name of file plus
```

Reply:
Oh, I didn't know, thanks)